pyaerocom/io/readgridded.py

```diff
@@ -1389,11 +1397,132 @@ def _eval_vert_which_and_ts_type(self, var_name, vert_which, ts_type):
                 ts_type = None
         return vert_which, ts_type
 
+    def check_constraint_valid(self, constraint):
+        if not isinstance(constraint, dict):
+            raise ValueError('Read constraint needs to be dict')
+        elif not 'operator' in constraint:
+            raise ValueError('Constraint requires specification of operator. '
+                             'Valid operators: {}'.format(self.CONSTRAINT_OPERATORS))
+        elif not constraint['operator'] in self.CONSTRAINT_OPERATORS:
+            raise ValueError('Invalid constraint operator. Choose from: {}'
+                             .format(self.CONSTRAINT_OPERATORS))
+        elif not 'filter_val' in constraint:
+            raise ValueError('constraint needs specification of filter_val')
+        elif not isnumeric(constraint['filter_val']):
+            raise ValueError('Need numerical filter value')
+
+    def apply_read_constraint(self, data, constraint,
+                              **kwargs):
+        """
+        Filter a `GriddeData` object by value in another variable
+
+        Note
+        ----
+        BETA version, that was hacked down in a rush to be able to apply
+        AOD>0.1 threshold when reading AE.
+
+
+        Parameters
+        ----------
+        data : GriddedData
+            data object to which constraint is applied
+        filter_var : TYPE
+            DESCRIPTION.
+        operator : str
+            operator used to filter other variable field to retrieve filter
+            mask
+        filter_val : float
+            value in filter_var for which mask will be retrieved
+        new_val : TYPE, optional
+            DESCRIPTION. The default is None.
+        **kwargs : TYPE
+            DESCRIPTION.
+
+        Raises
+        ------
+        ValueError
+            DESCRIPTION.
+
+        Returns
+        -------
+        None.
+
+        """
+        self.check_constraint_valid(constraint)
+        if 'new_val' in constraint:
+            new_val = constraint['new_val']
+        else:
+            new_val = np.ma.masked
```

Comment:
Is `new_val` always meant to be np.masked, np.nan or similar, or can it also be some finite value?

Reply:
The idea was that it can be any value